### PR TITLE
Enable deprecation consistency

### PR DESCRIPTION
**Pull request by Piinks**
This enables the deprecation consistency lint. :)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -105,7 +105,7 @@ linter:
     # - constant_identifier_names # needs an opt-out https://github.com/dart-lang/linter/issues/204
     - control_flow_in_finally
     # - curly_braces_in_flow_control_structures # not required by flutter style
-    # - deprecated_consistency # not yet tested
+    - deprecated_consistency # not yet tested
```
Comment (Piinks) [inline analysis_options.yaml:108]:
```suggestion
    - deprecated_consistency
```

**Review by a14n — DISMISSED** [review]
LGTM

**Comment by a14n** [comment]
It looks like linter-1.0.0 (containing this new lint) has not yet been rolled into the sdk https://github.com/dart-lang/sdk/blob/e7644d9a4c6f400680206f6d03135f24d0480052/DEPS#L121

**Review by a14n — COMMENTED** [review]
It looks like linter-1.0.0 (containing this new lint) has not yet been rolled into the sdk https://github.com/dart-lang/sdk/blob/e7644d9a4c6f400680206f6d03135f24d0480052/DEPS#L121

**Comment by goderbauer** [comment]
I think the new lint has now been rolled into the SDK and Flutter...

**Comment by Piinks** [comment]
PTAL, I think this is ready now. :)

**Review by a14n — APPROVED** [review]
LGTM

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**